src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,398 @@
+
+        Args:
+            ior_results (dict): Dictionary that contains the IOR run results.
+            job_num (int): Job number used for the IOR run.
+            errors (list): Error list used in the test.
+        """
+        self.log.info(ior_results[job_num])
+        if not ior_results[job_num][0]:
+            ior_error = ior_results[job_num][1]
+            errors.append(f"Error found in second IOR run! {ior_error}")
+
+    def create_ip_to_host(self):
+        """Create a dictionary of IP address to hostname of the server nodes.
+        """
+        command = "hostname -i"
+        results = run_pcmd(hosts=self.hostlist_servers, command=command)
+        self.log.info("hostname -i results = %s", results)
+
+        return {result["stdout"][0]: str(result["hosts"]) for result in results}
+
+    def verify_network_failure(self, ior_namespace, container_namespace):
+        """Verify network failure can be recovered without intervention in DAOS side.
+
+        1. Create a pool and a container. Create a container with or without redundancy
+        factor based on container_namespace.
+        2. Take down network interface of one of the engines, say ib0 of rank 0. hsn0 in
+        Aurora.
+        3. Run IOR with given object class.
+        4. Bring up the network interface.
+        5. Restart DAOS with dmg.
+        6. Call dmg pool query -b to find the disabled ranks.
+        7. Call dmg pool reintegrate --rank=<rank> one rank at a time to enable all
+        ranks. Wait for rebuild after calling the command.
+        8. Run IOR again. It should work this time.
+        9. To further verify the system, create another container.
+        10. Run IOR to the new container. Should work.
+
+        Note that I'm not sure about the usefulness of testing different object classes
+        and redundancy factors. We probably have to understand how data are exchanged
+        among the ranks based on the object class.
+
+        Args:
+            ior_namespace (str): Yaml namespace that defines the object class used for
+                IOR.
+            container_namespace (str): Yaml namespace that defines the container
+                redundancy factor.
+        """
+        # 1. Create a pool and a container.
+        self.container = []
+        self.add_pool(namespace="/run/pool_size_ratio_80/*")
+        self.container.append(
+            self.get_container(pool=self.pool, namespace=container_namespace))
+
+        # 2. Take down network interface of one of the engines. Use the first host.
+        errors = []
+        self.network_down_host = self.hostlist_servers[0]
+        self.log.info("network_down_host = %s", self.network_down_host)
+        self.interface = self.params.get(
+            "fabric_iface", "/run/server_config/servers/0/*")
+        self.log.info("interface to update = %s", self.interface)
+
+        if self.test_env == "ci":
+            # wolf
+            self.update_network_interface(
```

Comment:
(pylint-no-member) Instance of 'NetworkFailureTest' has no 'update_network_interface' member